autoload/ale/hover.vim

```diff
@@ -95,34 +108,45 @@ function! s:ShowDetails(linter) abort
     let l:root = l:lsp_details.project_root
 
     if a:linter.lsp is# 'tsserver'
+        let l:column = a:column
+
         let l:message = ale#lsp#tsserver_message#Quickinfo(
-        \   l:buffer,
-        \   l:line,
+        \   a:buffer,
+        \   a:line,
         \   l:column
         \)
     else
         " Send a message saying the buffer has changed first, or the
         " hover position probably won't make sense.
-        call ale#lsp#Send(l:id, ale#lsp#message#DidChange(l:buffer), l:root)
+        call ale#lsp#Send(l:id, ale#lsp#message#DidChange(a:buffer), l:root)
 
-        let l:column = min([l:column, len(getline(l:line))])
+        let l:column = min([a:column, len(getbufline(a:buffer, a:line)[0])])
 
-        let l:message = ale#lsp#message#Hover(l:buffer, l:line, l:column)
+        let l:message = ale#lsp#message#Hover(a:buffer, a:line, l:column)
     endif
 
     let l:request_id = ale#lsp#Send(l:id, l:message, l:root)
 
     let s:hover_map[l:request_id] = {
-    \   'buffer': l:buffer,
-    \   'line': l:line,
+    \   'buffer': a:buffer,
+    \   'line': a:line,
     \   'column': l:column,
+    \   'hover_from_balloonexpr': a:called_from_balloonexpr,
     \}
 endfunction
 
-function! ale#hover#Show() abort
-    for l:linter in ale#linter#Get(&filetype)
+" Obtain Hover information for the specified position
+" Only 1 optional argument is expected :
+"   - called_from_balloonexpr, this flag marks if we want the result from this
+"     ale#hover#Show to display in a balloon if possible
+"
+" Currently, the callbacks displays the info
+" - in the balloon if requested from balloonexpr and balloon_show is detected
+" - as status message otherwise
+function! ale#hover#Show(buffer, line, col, ...) abort
```

Comment:
Keeping track of the arguments will probably get a bit tedious as we go on. I think it's probably best to just write a single argument here instead, which contains all of the arguments in on `Dictionary`. That's probably the most future-proof option.

Reply:
Optional arguments is my (lazy) way of making changes without having to delve too much in changing the tests to make them work, this is definitely something I want to make better. I'll implement the dictionary, and call it opt (for optional arguments).

The `get()` function should help with default values anyway.